### PR TITLE
overhaul project Readme

### PR DESCRIPTION
**Pull request by titusfortner**
This started as documenting the Windows build steps now that you don't have to do the weird visual studio version workaround.

Then it turned into reorganizing the sections.

That turned into updating a bunch of things in the Rakefile.

I figured I'd ask for someone else to look at it before I just merged it.  Thanks.

Hmm, I'm already seeing a few things I want to tweak more, but I've spent so much time on this already, lol... Let me know what you think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,194 +15,381 @@ generously donated thousands of hours in code development and upkeep.
...
+If you are looking at getting started using selenium check out our [Getting Started Guide](https://www.selenium.dev/documentation/webdriver/getting_started/).
+
+### All Platforms
 * [Bazelisk](https://github.com/bazelbuild/bazelisk), a Bazel wrapper that automatically downloads
   the version of Bazel specified in `.bazelversion` file and transparently passes through all
   command-line arguments to the real Bazel binary.
-* Java JDK version 11 or greater (e.g., [Java 11 OpenJDK](https://openjdk.java.net/))
-* `java` and `jar` on the `$PATH` (make sure you use `java` executable from JDK but not JRE).
+* Java JDK version 17 or greater (e.g., [Java 17 Temurin](https://adoptium.net/temurin/releases/?version=17))
+  * Set `JAVA_HOME` environment variable to location of javac.exe
```
Comment (diemol) [inline README.md]:
```suggestion
  * Set `JAVA_HOME` environment variable to location of the Java binary
```

File: README.md [hunk 2]
```diff
@@ -15,194 +15,381 @@ generously donated thousands of hours in code development and upkeep.
...
+  * Rosetta for Apple Silicon Macs. Add `build --host_platform=//:rosetta` to the `.bazelrc.local` file. We are working
   to make sure this isn't required in the long run.
-* Windows users:
-  *  Latest version of [Visual Studio](https://www.visualstudio.com/) with command line tools and build tools installed
-  * A setup guide with detailed explanations can be seen on Jim Evan's [post](http://jimevansmusic.blogspot.com/2020/04/setting-up-windows-development.html)
-  * An up-to-date list of instructions for Windows 11, including avoiding issues with the latest versions of Visual Studio, can be seen in this [gist](https://gist.github.com/titusfortner/aec103e9b02709f771497fdb8b21154c)
 
-### Internet Explorer Driver
+### Windows
+Several years ago Jim Evans published a great article on
```
Comment (diemol) [inline README.md]: Linking his Twitter or LinkedIn profile would be nice.

**Review by diemol — CHANGES_REQUESTED** [review]
A couple of minor suggestions, thanks @titusfortner!

**Review by diemol — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -15,194 +15,335 @@ generously donated thousands of hours in code development and upkeep.
...
+    * Add Visual C++ build tools installation directory location to `BAZEL_VC` environment variable (e.g. `"C:\Program Files\Microsoft Visual Studio\2022\Community\VC"`)
+    * Add Visual C++ Build tools version to `BAZEL_VC_FULL_VERSION` environment variable (this can be discovered from the directory name in `"$BAZEL_VC\Tools\MSVC\<BAZEL_VC_FULL_VERSION>"`)
+5. Add support for long file names (bazel has a lot of nested directories that can exceed default limits in Windows)
+    * Enable Long Paths support with these 2 registry commands:
+    ```shell
+    reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Command Processor" /t REG_DWORD /f /v "DisableUNCCheck" /d "1"
+    reg add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem" /t REG_DWORD /f /v "LongPathsEnabled" /d "1"
+    ```
+    * Allow Bazel to create short name versions of long file paths: `fsutil 8dot3name set 0`
+    * Set bazel output to `C:/tmp` instead of nested inside project directory:
```
Comment (p0deje) [inline README.md:87]: Should we just commit this to the repo?

**Review by p0deje — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (titusfortner) [reply README.md:87]: Can we have a conditional toggle based on whether the user is on windows? How do we know they have access to their `C:\` drive even if on Windows?

Comment (p0deje) [reply README.md:87]: Right, we cannot use conditionals on startup options, so disregard.

**Comment by titusfortner** [comment]
I'm merging this. Let me know if I need to revert, better explain or add things that are missing.

File: README.md [hunk 4]
```diff
@@ -15,194 +15,335 @@ generously donated thousands of hours in code development and upkeep.
...
+## Contributing
 
 Please read [CONTRIBUTING.md](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md)
 before submitting your pull requests.
 
-## Requirements
 
+## Installing
+
+These are the requirements to cereate your own local dev environment to contribute to Selenium.
```
Comment (apupier) [inline README.md:31]: typo cereate --> create

**Pull request merged**